src/main/java/de/tum/in/www1/artemis/service/iris/IrisRateLimitService.java

```diff
@@ -0,0 +1,64 @@
+package de.tum.in.www1.artemis.service.iris;
+
+import java.time.ZonedDateTime;
+import java.util.Objects;
+
+import org.springframework.context.annotation.Profile;
+import org.springframework.stereotype.Service;
+
+import de.tum.in.www1.artemis.domain.User;
+import de.tum.in.www1.artemis.repository.iris.IrisMessageRepository;
+
+/**
+ * Service for the rate limit of the iris chatbot.
+ */
+@Service
+@Profile("iris")
+public class IrisRateLimitService {
+
+    private final IrisMessageRepository irisMessageRepository;
+
+    private final IrisSettingsService irisSettingsService;
+
+    public IrisRateLimitService(IrisMessageRepository irisMessageRepository, IrisSettingsService irisSettingsService) {
+        this.irisMessageRepository = irisMessageRepository;
+        this.irisSettingsService = irisSettingsService;
+    }
+
+    /**
+     * Get the rate limit information for the given user.
+     * See {@link IrisRateLimitInformation} and {@link IrisRateLimitInformation#isRateLimitExceeded()} for more information.
+     *
+     * @param user the user
+     * @return the rate limit information
+     */
+    public IrisRateLimitInformation getRateLimit(User user) {
+        var globalSettings = irisSettingsService.getGlobalSettings();
+        var irisChatSettings = globalSettings.getIrisChatSettings();
+        var start = ZonedDateTime.now().minusHours(3);
+        var end = ZonedDateTime.now();
+        var currentRateLimit = irisMessageRepository.countLlmResponsesOfUserWithinTimeframe(user.getId(), start, end);
```

Comment:
Is this really a limit? Is that not simply the recent amount of requests?

Reply:
Yes it is the amount if requests within the rate-limit timeframe. I know the variable naming isn't ideal, but I couldn't think of anything better yesterday 😅 
I'll see if I can find a better name (maybe "currentMessageCount").

Reply:
Renamed the entire `IrisRateLimitInformation` related stuff to better represent what it does.